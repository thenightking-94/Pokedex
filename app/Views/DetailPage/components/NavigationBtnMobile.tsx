import { useRouter } from "next/router"
import MobileBackIcon from "../../Icons-svg/mobileBackIcon"
import MobileForwardIcon from "../../Icons-svg/mobileForwardIcon"


export default function NavigationBtnMobile ({id}){
    const router = useRouter()

    return(
        <div className="nav-mobile">
            <div onClick={()=>{router.push(`/detail/${Number(id)-1}`)}} className='back-icon'><MobileBackIcon/>Previous</div>
            <div onClick={()=>{router.push(`/detail/${Number(id)+1}`)}} className='forward-icon'>Next<MobileForwardIcon/></div>
        </div>
    )
}