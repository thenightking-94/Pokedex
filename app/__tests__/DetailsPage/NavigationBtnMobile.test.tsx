import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import renderer from "react-test-renderer";
import NavigationBtnMobile from '../../Views/DetailPage/components/NavigationBtnMobile'

const queryClient = new QueryClient()

const props = {
                id: '6'
              }

describe('Navigation Btn', () => {
  it('renders correctly', () => {
    render(
    <QueryClientProvider client={queryClient}>
         <NavigationBtnMobile {...props} />
    </QueryClientProvider>
   )
  })
})

describe("Jest Snapshot testing suite", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create( 
      <QueryClientProvider client={queryClient}>
        <NavigationBtnMobile {...props}/>
     </QueryClientProvider>).toJSON();
      expect(domTree).toMatchSnapshot();
    });
  });