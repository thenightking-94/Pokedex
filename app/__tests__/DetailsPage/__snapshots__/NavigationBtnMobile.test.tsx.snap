// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="nav-mobile"
>
  <div
    className="back-icon"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="11"
      viewBox="0 0 14 11"
      width="14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.35356 1.85343L5.70711 1.49988L5.00001 0.792771L4.64645 1.14632L5.35356 1.85343ZM1.00001 5.49988L0.646454 5.14632L0.292901 5.49988L0.646454 5.85343L1.00001 5.49988ZM4.64645 9.85343L5.00001 10.207L5.70711 9.49988L5.35356 9.14632L4.64645 9.85343ZM4.64645 1.14632L0.646454 5.14632L1.35356 5.85343L5.35356 1.85343L4.64645 1.14632ZM0.646454 5.85343L4.64645 9.85343L5.35356 9.14632L1.35356 5.14632L0.646454 5.85343ZM1.00001 5.99988L14 5.99988L14 4.99988L1.00001 4.99988L1.00001 5.99988Z"
        fill="white"
      />
    </svg>
    Previous
  </div>
  <div
    className="forward-icon"
    onClick={[Function]}
  >
    Next
    <svg
      fill="none"
      height="11"
      viewBox="0 0 14 11"
      width="14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.64645 9.14645L8.29289 9.5L9 10.2071L9.35355 9.85355L8.64645 9.14645ZM13 5.5L13.3536 5.85355L13.7071 5.5L13.3536 5.14645L13 5.5ZM9.35355 1.14645L9 0.792893L8.29289 1.5L8.64645 1.85355L9.35355 1.14645ZM9.35355 9.85355L13.3536 5.85355L12.6464 5.14645L8.64645 9.14645L9.35355 9.85355ZM13.3536 5.14645L9.35355 1.14645L8.64645 1.85355L12.6464 5.85355L13.3536 5.14645ZM13 5H0V6H13V5Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;
